From 2f9b155e51214bef9f9046a14b02b2c96a270030 Mon Sep 17 00:00:00 2001
From: Sekhar Nori <nsekhar@ti.com>
Date: Sun, 24 Nov 2013 20:43:09 +0530
Subject: [PATCH 172/285] ARM: dts: am43x-epos-evm: add support for clkout2

Add support for clkout2 pin on AM43x ePOS EVM. On
this EVM this pin provides 32khz clock to WLAN module.

Signed-off-by: Sekhar Nori <nsekhar@ti.com>
Signed-off-by: Kishon Vijay Abraham I <kishon@ti.com>
[Kevin: The original patch taken from TI
ti-sdk-am437x-evm-07.01.00.00-Linux-x86-Install.bin]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm/boot/dts/am43x-epos-evm.dts |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/am43x-epos-evm.dts b/arch/arm/boot/dts/am43x-epos-evm.dts
index ae005e1..89ab0c0 100644
--- a/arch/arm/boot/dts/am43x-epos-evm.dts
+++ b/arch/arm/boot/dts/am43x-epos-evm.dts
@@ -153,6 +153,8 @@
 };
 
 &am43xx_pinmux {
+	pinctrl-names = "default";
+	pinctrl-0 = <&dss_pinctrl &clkout2_pin>;
 
 	cpsw_default: cpsw_default {
 		pinctrl-single,pins = <
@@ -317,6 +319,12 @@
 		>;
 	};
 
+	clkout2_pin: clkout2_pin {
+		pinctrl-single,pins = <
+			0x274 (PIN_OUTPUT_PULLUP | MUX_MODE3)
+		>;
+	};
+
 	wlirq_pins: pinmux_wlirq_pins {
 		pinctrl-single,pins = <
 			0x74 (PIN_INPUT | WAKEUP_ENABLE | MUX_MODE7)   /*gpmc_wpn.gpio0_31 */
@@ -746,3 +754,7 @@
 		};
 	};
 };
+
+&clkout_32k_mux_ck {
+        clock-default =  <&clk_32k_tpm_ck>;
+};
-- 
1.7.5.4

